src/coffee/export.coffee

```diff
@@ -112,14 +113,21 @@ class Export
   # return the correct product service in case query string is used or not
   _getProductService: (staged = true, customWherePredicate = false) ->
     productsService = @client.productProjections
-    if customWherePredicate
-      productsService.where(customWherePredicate)
+    perPage = 100
 
     if @queryOptions.queryString
-      productsService.byQueryString(@queryOptions.queryString, @queryOptions.isQueryEncoded)
+      query = @queryOptions.queryString
+      if @queryOptions.isQueryEncoded
+        query = decodeURIComponent(query)
+
+      if customWherePredicate
+        query = @_appendPredicateToQueryString(query, customWherePredicate)
+
+      productsService.byQueryString(query, false)
       productsService
     else
-      productsService.all().perPage(500).staged(staged)
+      productsService.where(customWherePredicate || '')
+      productsService.all().perPage(perPage).staged(staged)
```

Comment:
there are used two ways how the products are loaded:
- when the queryString is not specified we just add `where` condition to filter products for specified productType when doing a full export
- when the queryString is present there is used `byQueryString` which ignored `where` so I added predicate (for filtering productType's products) directly to queryString